File: PBO/date/Date.java
public class Date {
   // Variabel instan pribadi
   private int year, month, day;

   public Date(int year, int month, int day) {
      this.year = year;
      this.month = month;
      this.day = day;
   }

   /** Mengembalikan year / tahun */
   public int getYear() {
      return this.year;
   }
   /** Mengembalikan month / bulan */
   public int getMonth() {
      return this.month;
   }
   /** Mengembalikan day / hari */
   public int getDay() {
      return this.day;
   }
   /** Menetapkan tahun */
   public void setYear(int year) {
      this.year = year;
   }
   /** Menetapkan bulan */
   public void setMonth(int month) {
      this.month = month;
   }
   /** Menetapkan hari */
   public void setDay(int day) {
      this.day = day;
   }

   /* Mengembalikan String deskriptif dalam bentuk "MM/DD/YYYY" dengan awalan nol */
   public String toString() {
        return String.format("%02d/%02d/%4d", month, day, year);
             
   }

   /** Menetapkan tahun, bulan dan hari. Tidak ada validasi masukan */
   public void setDate(int year, int month, int day) {
      this.year = year;
      this.month = month;
      this.day = day;
   }
}

File: PBO/date c++/Date.cpp
#include "Date.h"
#include <iostream>
#include <iomanip>

Date::Date(int year, int month, int day)
    : year(year), month(month), day(day) {}

int Date::getYear() const {
    return year;
}

int Date::getMonth() const {
    return month;
}

int Date::getDay() const {
    return day;
}

void Date::setYear(int year) {
    this->year = year;
}

void Date::setMonth(int month) {
    this->month = month;
}

void Date::setDay(int day) {
    this->day = day;
}

void Date::setDate(int year, int month, int day) {
    this->year = year;
    this->month = month;
    this->day = day;
}

std::string Date::toString() const {
    std::ostringstream oss;
    oss << std::setfill('0') << std::setw(2) << month << "/"
        << std::setw(2) << day << "/" << std::setw(4) << year;
    return oss.str();
}
